packages/caliper-core/test/manager/monitors/monitor-docker.js

```diff
@@ -0,0 +1,211 @@
+            dockerMonitor.interval.should.equal(monitorOptions.interval*1000);
+        });
+
+        it('should set interval to default if not provided in options', () => {
+            dockerMonitor = new DockerMonitorRewire(emptyOptions);
+            dockerMonitor.interval.should.equal(ConfigUtil.get(ConfigUtil.keys.Monitor.Interval));
+        });
+    });
+
+    describe('when finding containers', () => {
+
+        beforeEach( () => {
+            dockerMonitor = new DockerMonitorRewire(monitorOptions);
+        });
+
+        it('should add containers objects to container field if exitents', async () => {
+            dockerMonitor.SystemInformation = SystemInformation;
+            await dockerMonitor.findContainers().then(()=>{
+                dockerMonitor.containers.should.be.an('array').that.have.lengthOf(monitorOptions.containers.length);
+            });
+        });
+
+        it('should add and label containers as local', async () => {
+            const containers = [
+                {
+                    id:"118",
+                    name:"peer0.org1.example.com",
+                    remote:null
+                }
+            ]
+            await dockerMonitor.findContainers().then(()=>{
+                dockerMonitor.containers.should.be.an('array').that.deep.equals(containers);
+            });
+        });
+
+        /*it('should add and label containers as remote', async () => {
+            await dockerMonitor.findContainers().then(()=>{
+                for(let key in this.options.containers){
+                    dockerMonitor.containers[i].name
+                }
+            });
+        });*/
+    });
+
+    describe('whan reading statistics', () => {
+
+        //memory test values
+        const mem_usage = dockerStats.memory_stats.usage - dockerStats.memory_stats.stats.cache;
+        const mem_percent = mem_usage / dockerStats.mem_limit;
+
+        beforeEach( () => {
+            dockerMonitor = new DockerMonitorRewire(monitorOptions);
+        });
+
+        it('should set isReading to false', async () => {
+            await dockerMonitor.findContainers().then(async () => {
+                await dockerMonitor.readContainerStats()
+            }).then(async () => {
+                dockerMonitor.isReading.should.be.false;
+            });
+        });
+
+        it('should read and add stats for each target container', async () => {
+
+            await dockerMonitor.findContainers().then(async () => {
+                await dockerMonitor.readContainerStats()
+            }).then(async () => {
+                dockerMonitor.stats["118"].mem_usage.should.be.an('array').that.deep.equals([mem_usage]);
+            });
+        });
+
+    });
+
+    describe('when getting statistics', () => {
+
+        beforeEach( () => {
+            dockerMonitor = new DockerMonitorRewire(monitorOptions);
+        });
+
+        const cpuCores = os.cpus().length;
```

Comment:
If this test was run on a single core machine it would pass even if the code for cpu normalisation had the division removed. If possible we should stub out the use of os, the danger here is that it could break other tests that rely on os and expect os to actually work.in rewire there are some features such as revert which looks like they can change things back. I think you will want to call revert in the after block of the whole test file